pkg/replicate/scheme.go

```diff
@@ -313,3 +364,69 @@ func (rs *replicationScheme) ensureObjectReplicated(ctx context.Context, objectN
 
 	return nil
 }
+
+type compactMetrics struct {
```

Comment:
Why do we need to recreate compact metrics in replicate?

Reply:
I was going through, comments on previous PRs raised for the same issue, and the maintainers there had mentioned that blocks need to be grouped first and then replicated concurrently.
I followed the approach for grouping them, but that also required compaction. Grouped the blocks together like we do in [the compactor itself](https://github.com/thanos-io/thanos/blob/b5560c1c6c32093d9933e510d3b6ce5ca133f4b4/pkg/compact/compact.go#L285)